pkg/ingester/ingester.go

```diff
@@ -1386,6 +1417,14 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 	var builder labels.ScratchBuilder
 	var nonCopiedLabels labels.Labels
 	for _, ts := range timeseries {
+		// The cost attribution value for the series
+		costLabel := i.limits.CostAttributionLabel(userID)
+		var costAttrib string
+		// when cost attribution label is set
+		if costLabel != "" {
+			costAttrib = i.costAttribution.UpdateAttributionTimestamp(userID, validation.AttributionValue(costLabel, userID, ts.Labels), time.Now())
```

Comment:
Here and everywhere the time should be the request's time, not `time.Now()`, for both consistency and saving some resources calling this (calling `time.Now()` has a cost)